base/cvd/cuttlefish/common/libs/utils/archive.cpp

```diff
@@ -115,9 +117,72 @@ Result<std::string> ExtractImage(const std::string& archive_filepath,
   return {result.front()};
 }
 
+namespace {
+
+Result<std::string> hardlinkRecursively(const std::string& src_dir,
```

Comment:
Should return `Result<std::vector<std::string>>` and avoid the out parameter since the string result is ignored anyways.